### PR TITLE
Add statement about possibility of source map rejecting in case of an invalid version field

### PR DESCRIPTION
**Pull request by JSMonk**
I'm not quite sure about `can be rejected` instead of `should be rejected,` but I feel there should be freedom for choosing such behavior for tooling, but it is discussable.
Want to hear your opinions @mitsuhiko @littledan  @jkup

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
My opinion here is that we should enforce rejection as if it was, for example, an encoding error. There is a risk that one day in the future there will be a new source map version, and we should avoid source map v3 tools to accidentally consume a source map v(future) and provide wrong-but-not-rejected results.

**Comment by bomsy** [comment]
Also should we be specific about backward compatibility as well? E.g Implementations with support for v4 should still accept versions v3.

**Comment by littledan** [comment]
How about "must"?

**Comment by littledan** [comment]
> Also should we be specific about backward compatibility as well? E.g Implementations with support for v4 should still accept versions v3.

I don't think we're going to do any more version number increments.

**Comment by mitsuhiko** [comment]
I would recommend to treat it as an error in the file, but I would not go as far as mandating a particular rejection, at least not for all consumers of a source map. It should however not be considered a valid source map.

(Which goes a little bit into what a tool should be doing for invalid source maps and quite frankly my recommendation would be to still try to apply it, even if there are issues with it.)

**Comment by JSMonk** [comment]
It sounds like a topic to discuss on one of the upcoming meeting.

File: source-map.bs [hunk 1]
```diff
@@ -168,7 +168,7 @@ for practical reasons, the order cannot be defined in many JSON generators and
 has never been enforced.
 
 <dfn><code>version</code></dfn> is the version field which is always the number
-`3` as an integer.
+`3` as an integer. Source map can be rejected in case of a value different from `3`.
```
Comment (littledan) [inline source-map.bs:171]: ...which *must always be* the number `3` as an integer. The source map *may* be rejected in case of a value different from `3`.

**Comment by jkup** [comment]
Going to merge this. If, in the future, we realize there are multiple invalid fields that consumers try to recover, we should consider making a top level note that consumers may try to recover invalid source  maps.

**Review by jkup — APPROVED** [review]

**Pull request merged**